Getting Error as "java.net.URISyntaxException: Relative path in absolute URI" while running my new project with spoon

Running an espresso test class with the following command:
```
gradlew spoon -PspoonClassName=com.me.kioskapp.testCases.ATC_Kiosk_002
 --stacktrace
```

Getting the URI Syntax Exception (java.net.URISyntaxException).

Error Details below:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:spoonDebugAndroidTest'.
> java.net.URISyntaxException: Relative path in absolute URI: file://F:%5CKiosk_App%5Ckiosk-41d308ec738c_spoon%5Ca
pp%5Cbuild%5Ccustom-report-dir%5Cdebug%5Cindex.html

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:spoonDebugAndroidTest'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter
.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:7
0)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachi
ngStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExe
cuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStat
eTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.
java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.
java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:3
4)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter
.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuild
OperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuild
OperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecut
er.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecut
er.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecuto
r.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor
.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.
java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.
java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:9
9)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.internal.UncheckedException: java.net.URISyntaxException: Relative path in absolute URI: file://F
:%5CKiosk_App%5Ckiosk-41d308ec738c_spoon%5Capp%5Cbuild%5Ccustom-report-dir%5Cdebug%5Cindex.html
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:63)
        at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:40)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:76)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTa
skClassInfoStore.java:141)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTask
ClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTask
ClassInfoStore.java:121)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:731)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:705)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122
)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuild
OperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuild
OperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.
java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter
.java:92)
        ... 27 more
Caused by: java.net.URISyntaxException: Relative path in absolute URI: file://F:%5CKiosk_App%5Ckiosk-41d308ec738c_
spoon%5Capp%5Cbuild%5Ccustom-report-dir%5Cdebug%5Cindex.html
        at com.jaredsburrows.spoon.SpoonTask.getClickableFileUrl(SpoonTask.kt:116)
        at com.jaredsburrows.spoon.SpoonTask.spoonTask(SpoonTask.kt:111)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        ... 39 more


* Get more help at https://help.gradle.org

BUILD FAILED in 37s
168 actionable tasks: 85 executed, 83 up-to-date

I did some google for the issue and it seems that /// (triple slashes) need to be used in the filename like this [file:///F:/Kiosk_App/kiosk-41d308ec738c_
spoon/app/build/custom-report-dir/debug/index.html
```

--- a/src/main/kotlin/com/jaredsburrows/spoon/ConsoleRenderer.kt
+++ b/src/main/kotlin/com/jaredsburrows/spoon/ConsoleRenderer.kt
@@ -1,0 +1,70 @@
+package com.jaredsburrows.spoon
+
+import org.gradle.api.UncheckedIOException
+import java.io.File
+import java.io.IOException
+import java.net.URI
+import java.net.URISyntaxException
+
+/**
+ * Renders information in a format suitable for logging to the console.
+ *
+ * Taken from: https://github.com/gradle/gradle/blob/master/subprojects/logging/src/main/java/org/gradle/internal/logging/ConsoleRenderer.java
+ */
+class ConsoleRenderer {
+  /**
+   * Renders a path name as a file URL that is likely recognized by consoles.
+   */
+  fun asClickableFileUrl(path: File): String {
+    // File.toURI().toString() leads to an URL like this on Mac: file:/reports/index.html
+    // This URL is not recognized by the Mac console (too few leading slashes). We solve
+    // this be creating an URI with an empty authority.
+    try {
+      return URI("file", "", path.toURI().path, null, null).toString()
+    } catch (e: URISyntaxException) {
+      throw UncheckedException.throwAsUncheckedException(e)
+    }
+  }
+}
+
+/**
+ * Wraps a checked exception. Carries no other context.
+ *
+ * Taken from: https://github.com/gradle/gradle/blob/master/subprojects/base-services/src/main/java/org/gradle/internal/UncheckedException.java
+ */
+class UncheckedException : RuntimeException {
+  constructor(cause: Throwable) : super(cause)
+  constructor(message: String, cause: Throwable) : super(message, cause)
+
+  companion object {
+    /**
+     * Note: always throws the failure in some form. The return value is to keep the compiler happy.
+     */
+    @JvmOverloads fun throwAsUncheckedException(
+      t: Throwable,
+      preserveMessage: Boolean = false
+    ): RuntimeException {
+      if (t is RuntimeException) {
+        throw t
+      }
+
+      if (t is Error) {
+        throw t
+      }
+
+      if (t is IOException) {
+        if (preserveMessage) {
+          throw UncheckedIOException(t.message.orEmpty(), t)
+        } else {
+          throw UncheckedIOException(t)
+        }
+      }
+
+      if (preserveMessage) {
+        throw UncheckedException(t.message.orEmpty(), t)
+      } else {
+        throw UncheckedException(t)
+      }
+    }
+  }
+}

--- a/src/main/kotlin/com/jaredsburrows/spoon/SpoonTask.kt
+++ b/src/main/kotlin/com/jaredsburrows/spoon/SpoonTask.kt
@@ -6,7 +6,6 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.tasks.TaskAction
 import java.io.File
-import java.net.URI
 import java.time.Duration
 
 open class SpoonTask : DefaultTask() {
@@ -109,9 +108,7 @@ open class SpoonTask : DefaultTask() {
 
     val success = if (testing) testValue else builder.build().run()
     if (!success && !extension.ignoreFailures) {
-      throw GradleException("Tests failed! See ${getClickableFileUrl(outputDir, "index.html")}")
+      throw GradleException("Tests failed! See ${ConsoleRenderer().asClickableFileUrl(File(outputDir, "index.html"))}")
     }
   }
-
-  private fun getClickableFileUrl(path: File, fileName: String): String = URI("file", "", File(path.toURI().path, fileName).path, null, null).toString()
 }

